Dockerfile

```diff
@@ -18,25 +22,22 @@ ADD tools/c7n_mailer /src/tools/c7n_mailer
 
 WORKDIR /src
 
-RUN adduser --disabled-login custodian
 RUN apt-get --yes update \
- && apt-get --yes install build-essential curl --no-install-recommends \
+ && apt-get --yes install build-essential curl python3-venv --no-install-recommends \
```

Comment:
alot of this can be unrolled into individual run commands to take advantage of build cache, ie do base installs, add source, then run. ie here's what i used locally, bonus its, let me see play around with provider optimized images.

```dockerfile

FROM debian:10 as build-env

WORKDIR /src

RUN adduser --disabled-login custodian
RUN apt-get --yes update
RUN apt-get --yes install build-essential curl python3-venv python3-dev --no-install-recommends
RUN python3 -m venv /usr/local
RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python3

# Transfer Custodian source into container by directory
# to minimize size
ADD pyproject.toml poetry.lock README.md /src/
ADD c7n /src/c7n/
RUN . /usr/local/bin/activate && $HOME/.poetry/bin/poetry install

# ADD tools/c7n_gcp /src/tools/c7n_gcp
# RUN . /usr/local/bin/activate && cd tools/c7n_gcp && $HOME/.poetry/bin/poetry install

ADD tools/c7n_azure /src/tools/c7n_azure
RUN . /usr/local/bin/activate && cd tools/c7n_azure && $HOME/.poetry/bin/poetry install

# ADD tools/c7n_kube /src/tools/c7n_kube
# RUN . /usr/local/bin/activate && cd tools/c7n_kube && $HOME/.poetry/bin/poetry install && cd ../..
```